### PR TITLE
 drivers: flash: stm32: expose stm32 option byte apis

### PR DESCRIPTION
**Pull request by fabiobaltieri**
The STM32 flash driver includes some APIs to lock/unlock/set the option bytes. Make these public so that an application can use them to tweak option bytes as needed, add a read api as well.

Allows doing stuff like

```c
        uint32_t optb = 0;
        ret = flash_ex_op(DEVICE_DT_GET(DT_NODELABEL(flash)), FLASH_STM32_EX_OP_OPTB_READ, 0, &optb);
        printk("%d %08x\n", ret, optb);
```

### CODE REVIEW

**Review by erwango — COMMENTED** [review]
Thanks @fabiobaltieri.

Looks fine, but however, I wonder if, for consistency sake, these APIs shouldn't be made available through already existing `flash_ex_op()` api, which is there to expose flash controller specific features ?

Agreed this would be slightly more complex to implement and likely consume more footprint, but I think consistency is key for users.

Alternatively, if there are good reasons not to use `flash_ex_op()`, let's make it explicit.

**Comment by fabiobaltieri** [comment]
> Looks fine, but however, I wonder if, for consistency sake, these APIs shouldn't be made available through already existing `flash_ex_op()` api, which is there to expose flash controller specific features ?

Oh that makes load of sense, yeah thanks for the pointer I'll move this back to draft and rework it.

**Comment by fabiobaltieri** [comment]
@erwango I've reworked it to have two ex op codes but I don't really like the resulting code if I can be totally honest. What do you think?

**Review by de-nordic — DISMISSED** [review]

File: drivers/flash/Kconfig.stm32 [hunk 1]
```diff
@@ -84,6 +84,14 @@ config FLASH_STM32_BLOCK_REGISTERS
 	  registers improves system security, because flash content (or
 	  protection settings) can't be changed even when exploit was found.
 
+config FLASH_STM32_OPTION_BYTES
+	bool "Extended operation for option bytes access"
+	select FLASH_HAS_EX_OP
+	default n
+	help
+	  Enables flash extended operations that can be used to write and write
+	  STM32 option bytes.
```
Comment (gautierg-st) [inline drivers/flash/Kconfig.stm32:93]:
```suggestion
	  Enables flash extended operations that can be used to write and read
	  STM32 option bytes.
```

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Small typo, otherwise LGTM

File: drivers/flash/Kconfig.stm32 (hunk 1 above)
Comment (fabiobaltieri) [reply drivers/flash/Kconfig.stm32]: see, if my github account got compromised you'll be able to tell immediately as soon as you see a PR of me without at least one typo

**Comment by fabiobaltieri** [comment]
ah, needs a rebase too

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**